src/operator/src/insert.rs

```diff
@@ -558,6 +575,17 @@ impl Inserter {
             }
         }
 
+        // add ttl zero regions to context
+        for table_name in table_name_to_ids.keys() {
+            let table = if let Some(table) = self.get_table(catalog, &schema, table_name).await? {
+                table
+            } else {
+                continue;
+            };
+            let table_info = table.table_info();
+            check_ttl_zero_table(ctx, &table_info);
```

Comment:
I see we add this check in a lot of places. It's easy to make mistakes. I think we should refactor the return value of `RowToRegion`, `TableToRegion`, and `StatementToRegion` to tell us which request is for immediate regions.

Reply:
> I see we add this check in a lot of places. It's easy to make mistakes. I think we should refactor the return value of `RowToRegion`, `TableToRegion`, and `StatementToRegion` to tell us which request is for immediate regions.

How about just filter those insert requests out in this step? Like rename `TableToRegion.convert` to `convert_no_imme_region` and etc.